### PR TITLE
feat: allow custom types with inner types

### PR DESCRIPTION
**Pull request by samuelmaddock**
I'm considering making some of the IPC types support generics for https://github.com/electron/rfcs/pull/8. While experimenting, I noticed we don't generate them for custom types.

### CODE REVIEW

File: src/utils.ts
Comment (MarshallOfSound) [inline src/utils.ts:259]: `innerTypes` is an array, we should technically do this for any number of inner types not just the first one. E.g. `Foo<T, R>` should be supported (I think the docs parser will read this)

Comment (samuelmaddock) [reply src/utils.ts]: Confirmed. I tested with docs-parser and received multiple innerTypes (nice work btw!)

**Review by MarshallOfSound — APPROVED** [review]

**Comment by MarshallOfSound** [comment]
This will be blocked on https://github.com/electron/electron/pull/43985 just FYI, but should be good to go if you have the CFA token merge at will. Otherwise ping me on Slack and I can merge / release it.

**Comment by ckerr** [comment]
https://github.com/electron/electron/pull/43985 is merged so it shouldn't be blocking this PR anymore. @samuelmaddock @MarshallOfSound is this PR ready to merge?

**Pull request merged**
